pkg/metrics/metrics.go

```diff
@@ -11,85 +11,79 @@ const (
 	Namespace = "tidb2dw"
 )
 
-type Metrics struct {
+var (
 	TableNumGauge             prometheus.Gauge
 	SnapshotTotalSizeCounter  *prometheus.CounterVec
 	SnapshotLoadedSizeCounter *prometheus.CounterVec
-	IncrementPeddingSizeGauge *prometheus.GaugeVec
+	IncrementPendingSizeGauge *prometheus.GaugeVec
 	IncrementLoadedCounter    *prometheus.CounterVec
 	TableVersionsCounter      *prometheus.CounterVec
 	ErrorCounter              *prometheus.CounterVec
-}
+)
 
-func NewMetrics() *Metrics {
-	m := Metrics{}
-	m.TableNumGauge = prometheus.NewGauge(
+func InitMetrics() {
+	TableNumGauge = prometheus.NewGauge(
 		prometheus.GaugeOpts{
 			Namespace: Namespace,
 			Name:      "table_num",
 			Help:      "number of replication tables",
 		})
-	m.SnapshotTotalSizeCounter = prometheus.NewCounterVec(
+	SnapshotTotalSizeCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: Namespace,
 			Name:      "snapshot_total_size",
 			Help:      "total file size of snapshot",
 		}, []string{"table"})
-	m.SnapshotLoadedSizeCounter = prometheus.NewCounterVec(
+	SnapshotLoadedSizeCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: Namespace,
 			Name:      "snapshot_loaded_size",
 			Help:      "loaded file size of snapshot",
 		}, []string{"table"})
-	m.IncrementPeddingSizeGauge = prometheus.NewGaugeVec(
+	IncrementPendingSizeGauge = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Namespace: Namespace,
 			Name:      "increment_pedding_size",
 			Help:      "pedding increment file size in object storage",
 		}, []string{"table"})
-	m.IncrementLoadedCounter = prometheus.NewCounterVec(
+	IncrementLoadedCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: Namespace,
 			Name:      "increment_loaded_count",
 			Help:      "loaded increment file size",
 		}, []string{"table"})
-	m.TableVersionsCounter = prometheus.NewCounterVec(
+	TableVersionsCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: Namespace,
 			Name:      "table_versions",
 			Help:      "table version of each table",
 		}, []string{"table"})
-	m.ErrorCounter = prometheus.NewCounterVec(
+	ErrorCounter = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Namespace: Namespace,
 			Name:      "error_count",
 			Help:      "Total error count during replicating",
 		}, []string{"table"})
-	return &m
-}
-
-func (m *Metrics) SetTableNum(v int) {
-	m.TableNumGauge.Set(float64(v))
 }
 
-func (m *Metrics) Register() {
-	prometheus.MustRegister(m.TableNumGauge)
-	prometheus.MustRegister(m.SnapshotTotalSizeCounter)
-	prometheus.MustRegister(m.SnapshotLoadedSizeCounter)
-	prometheus.MustRegister(m.IncrementPeddingSizeGauge)
-	prometheus.MustRegister(m.IncrementLoadedCounter)
-	prometheus.MustRegister(m.TableVersionsCounter)
-	prometheus.MustRegister(m.ErrorCounter)
+func Register() {
```

Comment:
`InitMetrics` and `Register` to `func init()`. Then these metrics never needs unregister :)